app/main/download_data.py

```diff
@@ -120,8 +130,45 @@ def get_outcome_checkboxes() -> dict[str, Any]:
     return outcome_checkboxes
 
 
-returns = {
-    "name": FormNames.RETURNS_PERIOD,
-    "quarter": (1, 2, 3, 4),
-    "year": ("2022/2023", "2023/2024"),
-}
+def generate_financial_years(start_date, end_date):
+    # Adjust the years for the financial year
+    min_year = start_date.year if start_date.month > 3 else start_date.year - 1
+    max_year = end_date.year if end_date.month > 3 else end_date.year - 1
+
+    # Generate the list of financial years
+    financial_years = [
+        "{}/{}".format(year, year + 1) for year in range(min_year, max_year + 1)
+    ]
+
+    return financial_years
+
+
+# TODO decide wether to implement this or leave all quarter options available
+def generate_quarters(start_date, end_date):
+    # Calculate which quarter the given min and max month resides in
+    start_quarter = (start_date.month - 1) // 3 + 1
+    end_quarter = (end_date.month - 1) // 3 + 1
+
+    quarter_options = [1, 2, 3, 4]
+
+    return quarter_options[min(start_quarter, end_quarter) - 1 :: 1]
+
+
+def get_returns() -> dict[str, Any]:
+    returns_data = get_checkbox_data("/returns")
+
```

Comment:
there needs to be a check like
```
if not returns_data:
    return {
"name": FormNames.RETURNS_PERIOD,
        "from-quarter": [1, 2, 3, 4],
        "to-quarter": [1, 2, 3, 4],
        "from-year": [],
        "to-year": [],
}
```
or similar.

This should never happen but we shouldn't `500` error because of it.